catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -95,46 +111,40 @@
 
   Processor systemSubjectBinder = new SystemSubjectBinder();
 
-  /**
-   * Constructs a monitor for a specific directory that will ingest files into the Content
-   * Framework.
-   *
-   * @param camelContext the Camel context to use across all Content Directory Monitors. Note that
-   *     if Apache changes this ModelCamelContext interface there is no guarantee that whatever DM
-   *     is being used (Blueprint in this case) will be updated accordingly.
-   */
-  public ContentDirectoryMonitor(CamelContext camelContext, AttributeRegistry attributeRegistry) {
-    this(
-        camelContext,
-        attributeRegistry,
-        20,
-        5,
-        Executors.newSingleThreadExecutor(
-            StandardThreadFactoryBuilder.newThreadFactory("contentDirectoryMonitorThread")));
-  }
-
   /**
    * Constructs a monitor that uses the given RetryPolicy while waiting for the content scheme, and
    * the given Executor to run the setup and Camel configuration.
    *
-   * @param camelContext the Camel context to use across all Content directory monitors.
+   * @param camelContext the Camel context to use across all Content directory monitors. * @param
+   * @param attributeRegistry {@link AttributeRegistry} to use to perform attribute lookup for
+   *     attribute overrides for in-place monitoring
    * @param maxRetries Policy for polling the 'content' CamelComponent. Specifies, for any content
    *     directory monitor, the number of times it will poll.
    * @param delayBetweenRetries Policy for polling the 'content' CamelComponent. Specifies, for any
    *     content directory monitor, the number of seconds it will wait between consecutive polls.
    * @param configurationExecutor the executor used to run configuration and updates.
+   * @param transformerCheckPeriod amount of time in milliseconds to check for available
+   *     InputTransformers
+   * @param transformerWaitTimeout amount of time in milliseconds to timeout while waiting for
+   *     InputTransformers
    */
   public ContentDirectoryMonitor(
       CamelContext camelContext,
       AttributeRegistry attributeRegistry,
       int maxRetries,
       int delayBetweenRetries,
-      Executor configurationExecutor) {
+      Executor configurationExecutor,
+      InputTransformerIds inputTransformerIds,
+      long transformerCheckPeriod,
```

Comment:
I think the setters with the `VisibleForTesting` is more declarative. It also allows you to change the scope of setting those values. Passing in constants through blueprint has always irked me. This may be more of a nit 😅

Reply:
I agree with passing constants with blueprint but I am not a fan of setters either. My preference would be another constructor that calls this one with hardcoded constants to be used by blueprint. The current constructor can then be annotated with @VisibleForTesting.

I am a proponent of marking things final if they are not supposed to be modified during the life of an object. It simplifies review by making it clear it won't be modified, it simplifies concurrency since it is set at construction time, and more.